client/src/Game/GamePage.js

```diff
@@ -5,10 +5,38 @@ import Rules from "./Rules";
 
 
 class Piece {
-    constructor(color, name) {
-        this.color = color;
-        this.name = name;
-        this.isTrapped = false;
+    constructor(color, rank, isTrapped, row, column) {
```

Comment:
We already pass the power rank to you guys in the JSON board object, can you use the board to extract that information into your objects instead?

Reply:
That is what's happening. Pieces are instantiated below in the newBoard() method. We need the address a name, given a power rank, to a piece so that we can render the piece correctly.